### PR TITLE
Don't clean generated files

### PR DESCRIPTION
**Pull request by bcreane**
## Description

`make clean` no longer removes generated files. In addition, we weren't generating the deepcopy-gen associated with `upgrade/migrator/clients/v1/k8s/custom`, so I've added a target for that.

There's a new `check-gen-files` target which checks there are no un-committed changes to the generated files. This is connected to the pre-existing `static-checks` target.

Addresses: https://github.com/projectcalico/libcalico-go/issues/795.

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,6 +29,10 @@ DOCKER_GO_BUILD := mkdir -p .go-pkg-cache && \
 
 VENDOR_REMADE := false
 
+.PHONY: semaphore-ci
+semaphore-ci: .deepcopy_gen
```
Comment (bcreane) [inline Makefile]: Oops, this should depend on test-containerized, not .deepcopy-gen

File: Makefile [hunk 2]
```diff
@@ -29,13 +29,17 @@ DOCKER_GO_BUILD := mkdir -p .go-pkg-cache && \
 
 VENDOR_REMADE := false
 
+.PHONY: semaphore-ci
+semaphore-ci: test-containerized
```
Comment (tomdee) [inline Makefile]: Why add a new target for this? You could just add it to static-checks

File: .gitignore [hunk 3]
```diff
@@ -8,3 +8,4 @@ junit.xml
 *.swp
 .generate_exes
 .generate_files
+.deepcopy_gen
```
Comment (tomdee) [inline .gitignore]: I don't think we need the .deepcopy_gen file

File: Makefile [hunk 4]
```diff
@@ -77,7 +81,7 @@ ut: vendor/.up-to-date
 
 .PHONY: test-containerized
 ## Run the tests in a container. Useful for CI, Mac dev.
-test-containerized: vendor/.up-to-date run-etcd run-kubernetes-master
+test-containerized: vendor/.up-to-date .deepcopy_gen run-etcd run-kubernetes-master
```
Comment (tomdee) [inline Makefile]: Does this need to be here? Or should it live next to the git diff check?

File: Makefile [hunk 5]
```diff
@@ -169,16 +173,19 @@ stop-etcd:
...
 	rm -rf vendor .go-pkg-cache
 
-clean-deepcopy-gen:
-	rm -f .deepcopy_gen
-	find $(TOP_SRC_DIR) -name zz_generated* -exec rm {} \;
-
+.PHONY: clean-bin
 clean-bin:
-	rm -rf $(BINDIR) .deepcopy_gen_exes
+	rm -rf $(BINDIR)
```
Comment (tomdee) [inline Makefile]: nit: I don't think we really need a separate clean-bin target. Just put the `rm -rf $(BINDIR)` directly under clean:

File: Makefile [hunk 6]
```diff
@@ -169,16 +173,19 @@ stop-etcd:
...
+
+.PHONY: clean-deepcopy-gen
+clean-deepcopy-gen:
```
Comment (tomdee) [inline Makefile]: If this is just a developer convenience (it's not from anywhere) then you should add a comment stating that.

File: Makefile [hunk 7]
```diff
@@ -215,13 +222,46 @@ $(BINDIR)/deepcopy-gen:
 	$(DOCKER_GO_BUILD) \
 		sh -c 'go build -o $@ $(LIBCALICO-GO_PKG)/vendor/k8s.io/code-generator/cmd/deepcopy-gen'
 
-# Regenerate all files if the gen exe(s) changed
-.deepcopy_gen: .deepcopy_gen_exes
-	# Generate deep copies
+./lib/upgrade/migrator/clients/v1/k8s/custom/zz_generated.deepcopy.go: \
```
Comment (tomdee) [inline Makefile]: Nice. I like having the actual target file as a Makefile target.

File: Makefile [hunk 8]
```diff
@@ -215,13 +222,46 @@ $(BINDIR)/deepcopy-gen:
...
+										./lib/upgrade/migrator/clients/v1/k8s/custom/doc.go \
```
Comment (tomdee) [inline Makefile]: Identation seems a little off, below as well

File: Makefile [hunk 9]
```diff
@@ -215,13 +222,46 @@ $(BINDIR)/deepcopy-gen:
...
+										./lib/upgrade/migrator/clients/v1/k8s/custom/globalbgpconfig.go \
```
Comment (tomdee) [inline Makefile]: Can we add a wildcard rather than needing to list all the files?

File: Makefile [hunk 10]
```diff
@@ -215,13 +222,46 @@ $(BINDIR)/deepcopy-gen:
...
 	$(DOCKER_GO_BUILD) \
 		sh -c '$(BINDIR)/deepcopy-gen \
 			--v 1 --logtostderr \
 			--go-header-file "./docs/boilerplate.go.txt" \
 			--input-dirs "$(LIBCALICO-GO_PKG)/lib/apis/v3" \
 			--bounding-dirs "github.com/projectcalico/libcalico-go" \
 			--output-file-base zz_generated.deepcopy'
+
+.PHONY: .deepcopy_gen
+.deepcopy_gen:	.deepcopy_gen_exes \
```
Comment (tomdee) [inline Makefile]: Yep, we really don't need either of these .files. Can you get rid of them?

File: Makefile (hunk 8 above)
Comment (tomdee) [reply Makefile]: Oh, I just saw your ?ts=4 comment. Maybe it's OK, then.

**Comment by bcreane** [comment]
closing + re-opening to work around semaphore blocked in "pending" state.

**Comment by tomdee** [comment]
Looks awesome

**Pull request merged**